rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -688,8 +787,22 @@ extern "C" rmw_node_t * rmw_create_node(
 #endif
 
   dds_qos_t * qos = dds_create_qos();
+  if (qos == nullptr) {
+    RCUTILS_LOG_ERROR_NAMED("rmw_cyclonedds_cpp", "rmw_create_node: Unable to create qos");
+    node_gone_from_domain_locked(did);
+    return nullptr;
+  }
   std::string user_data = get_node_user_data(name, namespace_);
   dds_qset_userdata(qos, user_data.c_str(), user_data.size());
+
+  if (security_options->enforce_security) {
```

Comment:
This needs to use the rmw_security_enforcement_policy_t enum:

```suggestion
  if (security_options->enforce_security==RMW_SECURITY_ENFORCEMENT_ENFORCE) {
```